File: 005/problem005.swift
#!/usr/bin/env xcrun swift

func greatestCommonDivisor(_ x: Int, _ y: Int) -> Int {
    var x = x
    var y = y
    while y != 0 {
        let mod = x % y
        x = y
        y = mod
    }
    return x
}

func leastCommonMultiple(_ x: Int, _ y: Int) -> Int {
    return x * y / greatestCommonDivisor(x, y)
}


let result = (1...20).reduce(1) { sum, x in
    leastCommonMultiple(sum, x)
}

print(result)

File: 005/problem005.rs
fn greatest_common_divisor(a: i64, b: i64) -> i64 {
    let mut a = a;
    let mut b = b;
    while b != 0 {
        let module = a % b;
        a = b;
        b = module;
    }
    return a;
}

fn least_common_multiple(a: i64, b: i64) -> i64 {
    a * b / greatest_common_divisor(a, b)
}

fn main() {
    let result = (1..20).fold(1, |acc, x| least_common_multiple(acc, x));
    println!("{}", result);
}
